content/core-features/presence-and-occupancy/topic-presence_ia-update.textile

```diff
@@ -0,0 +1,737 @@
+  channel.presence.subscribe('enter', function(member) {
+    alert('Member ' + member.clientId + ' entered');
+  });
+  channel.presence.enter();
+```
+
+```[nodejs](code-editor:realtime/presence-simple)
+  const Ably = require('ably');
+  const realtime = new Ably.Realtime({
+    key: '{{API_KEY}}',
+    clientId: 'bob' }
+  );
+  const channel = realtime.channels.get('{{RANDOM_CHANNEL_NAME}}');
+  channel.presence.subscribe('enter', function(member) {
+    console.log('Member ' + member.clientId + ' entered');
+  });
+  channel.presence.enter();
+```
+
+```[ruby]
+  realtime = Ably::Realtime.new(key: '{{API_KEY}}', client_id: 'bob')
+  channel = realtime.channels.get('{{RANDOM_CHANNEL_NAME}}')
+  channel.presence.subscribe(:enter) do |member|
+    puts "Member #{member.client_id} entered"
+  end
+  channel.presence.enter
+```
+
+```[java]
+  ClientOptions options = new ClientOptions("{{API_KEY}}");
+  options.clientId = "bob";
+  AblyRealtime realtime = new AblyRealtime(options);
+  Channel channel = realtime.channels.get("{{RANDOM_CHANNEL_NAME}}");
+  channel.presence.subscribe(new PresenceListener() {
+    @Override
+    public void onPresenceMessage(PresenceMessage member) {
+      System.out.println("Member " + member.clientId + " : " + member.action.toString());
+    }
+  });
+  channel.presence.enter(null, new CompletionListener());
+```
+
+```[csharp]
+  ClientOptions options = new ClientOptions("{{API_KEY}}") { ClientId =  "bob"};
+  AblyRealtime realtime = new AblyRealtime(options);
+  IRealtimeChannel channel = realtime.Channels.Get("{{RANDOM_CHANNEL_NAME}}");
+  channel.Presence.Subscribe(member => {
+    Console.WriteLine("Member " + member.ClientId + " : " + member.Action);
+  });
+  await channel.Presence.EnterAsync(null);
+```
+
+```[objc]
+  ARTClientOptions *options = [[ARTClientOptions alloc] initWithKey:@"{{API_KEY}}"];
+  options.clientId = @"bob";
+  ARTRealtime *realtime = [[ARTRealtime alloc] initWithOptions:options];
+  ARTRealtimeChannel *channel = [realtime.channels get:@"{{RANDOM_CHANNEL_NAME}}"];
+  [channel.presence subscribe:ARTPresenceEnter callback:^(ARTPresenceMessage *member) {
+      NSLog(@"Member %@ entered", member.clientId);
+  }];
+  [channel.presence enter:nil];
+```
+
+```[swift]
+  let options = ARTClientOptions(key: "{{API_KEY}}")
+  options.clientId = "bob"
+  let realtime = ARTRealtime(options: options)
+  let channel = realtime.channels.get("{{RANDOM_CHANNEL_NAME}}")
+  channel.presence.subscribe(.enter) { member in
+      print("Member \(member.clientId) entered")
+  }
+  channel.presence.enter(nil)
+```
+
+<span lang="ruby">Note that all examples above assume you are running them within an EventMachine reactor. Find out more in our "Realtime usage documentation":/realtime/usage.</span>
+
+If you would prefer to just dive into code and see some examples of how to use presence via the Realtime API, then we recommend you take a look at our "Realtime tutorials":/tutorials.
```

Comment:
Minor style points - I would not use "dive into code". Also, use "Ably recommends" rather than "we".